Src/Base/AMReX_MultiFabUtil.H

```diff
@@ -996,26 +997,45 @@ Vector<typename MF::value_type> get_cell_data (MF const& mf, IntVect const& cell
 }
 
 template <typename MF, std::enable_if_t<IsFabArray<MF>::value,int> FOO>
-MF get_line_data (MF const& mf, int dir, IntVect const& cell)
+MF get_line_data (MF const& mf, int dir, IntVect const& cell, Box const& bnd_bx)
 {
+    bool do_bnd = (!bnd_bx.isEmpty());
+
     BoxArray const& ba = mf.boxArray();
     DistributionMapping const& dm = mf.DistributionMap();
     const auto nboxes = static_cast<int>(ba.size());
 
     BoxList bl(ba.ixType());
     Vector<int> procmap;
     Vector<int> index_map;
-    for (int i = 0; i < nboxes; ++i) {
-        Box const& b = ba[i];
-        IntVect lo = cell;
-        lo[dir] = b.smallEnd(dir);
-        if (b.contains(lo)) {
-            IntVect hi = lo;
-            hi[dir] = b.bigEnd(dir);
-            Box b1d(lo,hi,b.ixType());
-            bl.push_back(b1d);
-            procmap.push_back(dm[i]);
-            index_map.push_back(i);
+    if (!do_bnd) {
+        for (int i = 0; i < nboxes; ++i) {
+            Box const& b = ba[i];
+            IntVect lo = cell;
+            lo[dir] = b.smallEnd(dir);
+            if (b.contains(lo)) {
+                IntVect hi = lo;
+                hi[dir] = b.bigEnd(dir);
+                Box b1d(lo,hi,b.ixType());
+                bl.push_back(b1d);
+                procmap.push_back(dm[i]);
+                index_map.push_back(i);
+            }
+        }
+    } else {
+        for (int i = 0; i < nboxes; ++i) {
+            Box const& b = ba[i];
+            IntVect lo = bnd_bx.smallEnd();
+            IntVect hi = bnd_bx.bigEnd();
+            if (!b.contains(lo)) { lo[dir] = b.smallEnd(dir); }
+            if (b.contains(lo)) {
+                if (!b.contains(hi)) { hi[dir] = b.bigEnd(dir); }
+                if (!bnd_bx.contains(hi)) { continue; }
+                Box b1d(lo,hi,b.ixType());
+                bl.push_back(b1d);
+                procmap.push_back(dm[i]);
+                index_map.push_back(i);
+            }
```

Comment:
Could you explain what you try to achieve, not what your code does? If I try harder, I might be able to understand the code. It seems that I need to check 3x3=9 cases depending the comparison results of various lo and hi to understand it does. But I suspect it's equivalent to the following 
```suggestion
            Box const& b1d = bnd_box & b;
            if (b1d.ok()) {
                bl.push_back(b1d);
                procmap.push_back(dm[i]);
                index_map.push_back(i);
            }
```

Reply:
I would like a multifab returned that contains data on the bounding box only. What you have commented above is a cleaner and more efficient way to achieve that. I will add it to the PR.